File: src/GestorDeTareas.Application/DTOs/CrearTareaDto.cs
using GestorDeTareas.Domain.Enums;

namespace GestorDeTareas.Application.DTOs;

public class CrearTareaDto
{
    public string Titulo { get; set; } = string.Empty;
    public DateTime FechaLimite { get; set; }
    public PrioridadTarea Prioridad { get; set; }
    public PilarVida Pilar { get; set; }
    public TipoTarea Tipo { get; set; }

    public string? Intencion { get; set; }
}

File: gestor-tareas-frontend/src/app/features/tareas/models/crear-tarea.dto.ts
import { PilarVida } from './pilar-vida.enum';
import { PrioridadTarea } from './prioridad-tarea.enum';
import { TipoTarea } from './tipo-tarea.enum';

export interface CrearTareaDto {
  titulo: string;
  fechaLimite: string;
  prioridad: PrioridadTarea;
  pilar: PilarVida;
  tipo: TipoTarea;
  intencion?: string | null;
}